src/__tests__/medications/ViewMedication.test.tsx

```diff
@@ -0,0 +1,380 @@
+import { Badge, Button } from '@hospitalrun/components'
+import { act } from '@testing-library/react'
+import format from 'date-fns/format'
+import { mount } from 'enzyme'
+import { createMemoryHistory } from 'history'
+import React from 'react'
+import { Provider } from 'react-redux'
+import { Router, Route } from 'react-router-dom'
+import createMockStore from 'redux-mock-store'
+import thunk from 'redux-thunk'
+
+import ViewMedication from '../../medications/ViewMedication'
+import * as ButtonBarProvider from '../../page-header/button-toolbar/ButtonBarProvider'
+import * as titleUtil from '../../page-header/title/useTitle'
+import TextFieldWithLabelFormGroup from '../../shared/components/input/TextFieldWithLabelFormGroup'
+import MedicationRepository from '../../shared/db/MedicationRepository'
+import PatientRepository from '../../shared/db/PatientRepository'
+import Medication from '../../shared/model/Medication'
+import Patient from '../../shared/model/Patient'
+import Permissions from '../../shared/model/Permissions'
+import { RootState } from '../../shared/store'
+
+const mockStore = createMockStore<RootState, any>([thunk])
+
+describe('View Medication', () => {
+  let history: any
+  const mockPatient = { fullName: 'test' }
+  const mockMedication = {
+    id: '12456',
+    status: 'draft',
+    patient: '1234',
+    medication: 'medication',
+    intent: 'order',
+    priority: 'routine',
+    quantity: { value: 1, unit: 'unit' },
+    notes: 'medication notes',
+    requestedOn: '2020-03-30T04:43:20.102Z',
+  } as Medication
+
+  let setButtonToolBarSpy: any
+  let titleSpy: any
+  let medicationRepositorySaveSpy: any
+  const expectedDate = new Date()
```

Comment:
I think we should move the spies and history into the setup function for initialization. if they are tested later, then we should return them from the setup function.

Reply:
Thanks @jackcmeyer i have refactored the duplicated code into a setup function where possible. May you take a look if it is now alright or if i may need to take another look.